pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/render.go

```diff
@@ -222,7 +236,11 @@ func (r *components) renderComponent(ctx context.Context, acc v1alpha2.Applicati
 		if err := setWorkloadInstanceName(traitDefs, w, c, existingWorkload); err != nil {
 			return nil, err
 		}
-	} else {
+	}
+	if isControlledByApp && !byHelmModule {
+		// the workload by Helm module doesn't follow the name rule in KubeVela
+		// KubeVela should not rename it
```

Comment:
The logic can be buggy and hard to maintian, can we use a `swith case` for it?

Reply:
sure, `switch case` is better here.